interface/billing/get_claim_file.php

```diff
@@ -20,19 +20,63 @@
 }
 
 $content_type = "text/plain";
-$claim_file_dir = $GLOBALS['OE_SITE_DIR'] . "/documents/edi/";
 
+// The key contains the filename
 $fname = $_GET['key'];
-$fname = preg_replace("[/]", "", $fname);
-$fname = preg_replace("[\.\.]", "", $fname);
-$fname = preg_replace("[\\\\]", "", $fname);
 
-if (strtolower(substr($fname, (strlen($fname) - 4))) == ".pdf") {
-    $content_type = "application/pdf";
+// Because of the way the billing tables are constructed (as of 2021)
+// We may not know exactly where the file is, so we need to try a couple
+// different places. This is mainly because the full path is not stored
+// in the database. Also, the file could have been generated with the
+// 'gen_x12_based_on_ins_co' global set to 'on' but if it was turned off,
+// we still want to be able to download the file. So, we have to do a bit
+// of searching.
+// The edi directory is the default location.
+
+// the loc, if set, may tell us where the file is
+$location = $_GET['location'];
```

Comment:
can't see where this is used. just want to make sure this isn't allowing setting a directory or anything like that

Reply:
@bradymiller no, it's more of a "hint" where to find the file... if the location is 'temp', we look in the globally-configured temp dir.